### PR TITLE
Optimisations and cleanups in the shader interpreter

### PR DESCRIPTION
**Pull request by yuriks**
- Replaces `std::stack` with `boost::static_vector` (much faster)
 - Replaces two references to u32 types with values (simpler, faster)
 - Makes program counter an index instead of a pointer (simpler, faster)

### CODE REVIEW

**Comment by archshift** [comment]
LGTM

File: src/video_core/vertex_shader.cpp [hunk 1]
```diff
@@ -53,7 +52,7 @@ struct VertexShaderState {
     };
 
     // TODO: Is there a maximal size for this?
-    std::stack<CallStackElement> call_stack;
+    boost::container::static_vector<CallStackElement, 16> call_stack;
```
Comment (neobrain) [inline src/video_core/vertex_shader.cpp]: This is limited to 16 elements, right? If so, I'm not seeing any bounds checks...

File: src/video_core/vertex_shader.cpp [hunk 2]
```diff
@@ -88,17 +87,15 @@ static void ProcessShaderCode(VertexShaderState& state) {
         }
 
         bool exit_loop = false;
-        const Instruction& instr = *(const Instruction*)state.program_counter;
-        const SwizzlePattern& swizzle = *(SwizzlePattern*)&swizzle_data[instr.common.operand_desc_id];
+        const Instruction instr = { program_code[state.program_counter] };
```
Comment (neobrain) [inline src/video_core/vertex_shader.cpp]: How is a copy faster here?

Comment (yuriks) [inline src/video_core/vertex_shader.cpp]: It's a u32. The pointer/reference is actually larger than that. :P

**Comment by neobrain** [comment]
LGTM

**Pull request merged**